libs/velo-external-db-core/src/service/schema.ts

```diff
@@ -11,62 +32,122 @@ export default class SchemaService {
+        const collections = (!collectionIds || collectionIds.length === 0) ? 
+            await this.storage.list() : 
+            await Promise.all(collectionIds.map(async(collectionName: string) => ({ id: collectionName, fields: await this.schemaInformation.schemaFieldsFor(collectionName) })))
+                
+        return { 
+            collection: this.formatCollections(collections)
+        }
     }
 
-    async listHeaders() {
-        const collections = await this.storage.listHeaders()
-        return { schemas: collections.map((collection) => asWixSchemaHeaders(collection)) }
+    async create(collection: Collection): Promise<CreateCollectionResponse> {                
+        await this.storage.create(collection.id, WixFormatFieldsToInputFields(collection.fields))
+        await this.schemaInformation.refresh()
+        return { collection }
     }
 
-    async find(collectionNames: string[]) {
-        const dbs: Table[] = await Promise.all(collectionNames.map(async(collectionName: string) => ({ id: collectionName, fields: await this.schemaInformation.schemaFieldsFor(collectionName) })))
-        const dbsWithAllowedOperations = this.appendAllowedOperationsTo(dbs)
+    async update(collection: Collection): Promise<UpdateCollectionResponse> {
+        await this.validateOperation(Create)
+        
+        // remove in the end of development
+        if (!this.storage.changeColumnType) {
+            throw new Error('Your storage does not support the new collection capabilities API')
+        }
 
-        return { schemas: dbsWithAllowedOperations.map( asWixSchema ) }
-    }
+        const collectionColumnsInRequest = collection.fields
+        const collectionColumnsInDb = await this.storage.describeCollection(collection.id)
 
-    async create(collectionName: string) {
-        await this.validateOperation(Create)
-        await this.storage.create(collectionName)
-        await this.schemaInformation.refresh()
-        return {}
-    }
+        const {
+            columnsToAdd,
+            columnsToRemove,
+            columnsToChangeType
+        } = compareColumnsInDbAndRequest(collectionColumnsInDb, collectionColumnsInRequest)
+
+        // Adding columns
+        if (columnsToAdd.length > 0) {
+            await this.validateOperation(AddColumn)
+        }
+        await Promise.all(columnsToAdd.map(async(field) => await this.storage.addColumn(collection.id, field)))
+        
+        // Removing columns
+        if (columnsToRemove.length > 0) {
+            await this.validateOperation(RemoveColumn)
+        }
+        await Promise.all(columnsToRemove.map(async(fieldName) => await this.storage.removeColumn(collection.id, fieldName)))
+
+        // Changing columns type
+        if (columnsToChangeType.length > 0) {
+            await this.validateOperation(ChangeColumnType)
+        }
+        await Promise.all(columnsToChangeType.map(async(field) => await this.storage.changeColumnType!(collection.id, field)))
```

Comment:
it will be better to change it to :
`this.storage.changeColumnType?.(collection.id, field)))`
until all the implementations would support this method.